### ISSUE TITLE
Get form-data as string for HTTP request

### ISSUE THREAD

**Issue by filips123** [form-data/form-data#414]
Is it possible to get raw string for HTTP request for HTTP request so I could manually post it to URL with different library?

Something like `formData.get()` which will return
```
Content-Type: multipart/form-data; boundary=----WebKitFormBoundaryePkpFF7tjBAqx29L

------WebKitFormBoundaryePkpFF7tjBAqx29L
Content-Disposition: form-data; name="MAX_FILE_SIZE"

100000
------WebKitFormBoundaryePkpFF7tjBAqx29L
Content-Disposition: form-data; name="uploadedfile"; filename="hello.o"
Content-Type: application/x-object

... contents of file goes here ...
------WebKitFormBoundaryePkpFF7tjBAqx29L--
```

**Comment by filips123** [comment]
I found some quick and temporary solution:

```js
let data = ''
for (var i = 0, len = formData._streams.length; i < len; i++) {
  if (typeof formData._streams[i] !== 'function') {
    data += formData._streams[i] + '\r\n'
  }
}
```

**Comment by JBtje** [comment]
The `\r\n` only needs to be added after the data. FormData takes care of the breaks after e.g. `Content-Disposition` or `Content-Type`.

The below code is not.... perfect. If you want to post data that has the exact same same header it will not work... but the odds of you doing that are small.
```js
let data = '';
let hrd  = formData_.getHeaders();
for( let i = 0, len = formData_._streams.length; i < len; i++ ) {

    if( typeof formData_._streams[i] !== 'function' ) {
        data += formData_._streams[i];

        if( typeof formData_._streams[i] !== 'string' || formData_._streams[i].substring( 0, hrd.length ) === hrd ) {
            data += '\r\n';
        }
    }
}
```

**Comment by filips123** [comment]
What if there are values above 0x7F?

**Comment by JBtje** [comment]
Because the above method turns the data into a string, it will also have an encoding. Encodings are disastrous for uploading files via multipart forms. For example, 0xFF is received as 3 Bytes on the server using the above method.

So how to solve it? no clue....

The proper way to upload multipart data is by not using any form of encoding, thus a raw Buffer would do. But i'm still testing if I can get it to work (instead of adding the above items to a string, add them to a buffer and send the buffer via Axios).

**Comment by damianobarbati** [comment]
Any news on this? :(

**Comment by JBtje** [comment]
> Any news on this? :(

yes, a little while ago [getBuffer()](https://github.com/form-data/form-data#buffer-getbuffer) method has been added, which contains the full content of the multipart/form-data.
To get the string, you can use [toString()](https://github.com/form-data/form-data#string-tostring), however that will only work as long as the content has characerts below 0x7F. (see above post)
